plasmapy/physics/dispersion/two_fluid.py

```diff
@@ -0,0 +1,127 @@
+# FUNCTION TO CALCULATE PHASE SPEEDS OF THE THREE BRANCHES OF TWO FLUID
+# DISPERSION RELATION (e.g. STRINGER JPP 1963, ROGERS PRL 2001,
+# and BELLAN JGR 2012)
+
+#
+#                   Tulasi Nandan Parashar
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def tfps(beta=0.6, speed=1., de2=0.000545, theta=0, wavenumber=1):
+    """
+    beta: Total plasma beta,
+    speed: Alfven speed based on mean field,
+    de2: me/mi,
+    theta: Angle of propagation in degrees
+    wavenumber: Wavenumber of interest in units of kdi
+
+    Output is frequencies of the roots and the phase speeds w/k
+    The roots are w[0]:Fast/Whistler, w[1]:Alfven/KAW, w[2]: Slow/Cyclotron
+    """
+
+    pi_angle = theta * (np.pi / 180.0)
+    sine = np.sin(pi_angle)
+    cosine = np.cos(pi_angle)
+    tangent = sine / cosine
+    cs = np.sqrt(beta / 2.0) * speed
+    di = 1
+    caksq = (cosine ** 2) * (speed ** 2)
+    cmsq = (speed ** 2) + (cs ** 2)
+    D = 1 + wavenumber ** 2 * de2
+
+    term1 = -(speed ** 2 / D + cs ** 2 + caksq * (1 + wavenumber ** 2 * di ** 2 / D) / D) * wavenumber ** 2
+    term2 = caksq * wavenumber ** 4 * (speed ** 2 / D + cs ** 2 + cs ** 2 * (1 + wavenumber ** 2 * di ** 2 / D)) / D
+    term3 = -(cs ** 2 * wavenumber ** 6 * caksq ** 2) / D ** 2
```

Comment:
There may be a typo here. The last term in eq (6) of Roger's PRL has `caksq` in the denominator, not in the numerator.